src/other/utils.jl

```diff
@@ -58,3 +58,40 @@ function gennames(n::Integer)
     return res
 end
 
+"""
+    flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+
+When column `veccol` of `df` has non-zero length (i.e. a vector), returns a data frame where each element 
+of `veccol` is flattened, row `i` of `df` will be duplicated according to the length of `df[i, veccol]`
+"""
+function Base.Iterators.flatten(df::AbstractDataFrame, veccol::Union{Integer, Symbol})
+    N = nrow(df)
+    lengths = Vector{Int64}(undef, nrow(df))
+
+    for i in 1:N
+        lengths[i] = length(df[i, veccol])
```

Comment:
maybe simply `length.(df[!, veccol])`?

Reply:
also what is the contract on `eltype` of `veccol` - simply that it has `length`, or it should be iterable, or it should be an `AbstractVector`?